storage/mysql/log_storage_test.go

```diff
@@ -1092,32 +972,46 @@ func createTestLeaves(n, startSeq int64) []trillian.LogLeaf {
 }
 
 // Convenience methods to avoid copying out "if err != nil { blah }" all over the place
-func commit(tx storage.LogTreeTX, t *testing.T) {
-	if err := tx.Commit(); err != nil {
-		t.Fatalf("Failed to commit tx: %v", err)
+func beginLogTx(s storage.LogStorage, logID int64, t *testing.T) storage.LogTreeTX {
+	tx, err := s.BeginForTree(context.Background(), logID)
+	if err != nil {
+		t.Fatalf("Failed to begin log tx: %v", err)
 	}
+	return tx
 }
 
-func beginLogTx(s storage.LogStorage, logID logIDAndTest, t *testing.T) storage.LogTreeTX {
-	tx, err := s.BeginForTree(context.Background(), logID.logID)
+type committableTX interface {
+	Commit() error
+}
 
-	if err != nil {
-		t.Fatalf("Failed to begin log tx: %v", err)
+func commit(tx committableTX, t *testing.T) {
+	if err := tx.Commit(); err != nil {
+		t.Errorf("Failed to commit tx: %v", err)
 	}
+}
 
-	return tx
+type closeableTX interface {
+	Rollback() error
+	IsOpen() bool
 }
 
-func failIfTXStillOpen(t *testing.T, op string, tx storage.LogTreeTX) {
-	if r := recover(); r != nil {
-		// Check for the test bailing with panic before testing for unclosed tx.
-		// debug.Stack() does the right thing and includes the original failure point
-		t.Fatalf("Panic in %s: %v %v", op, r, string(debug.Stack()))
+// TODO(codingllama): Consider providing Close() methods to all TX types.
+func closeTX(tx closeableTX) error {
```

Comment:
You didn't misunderstood failIfTXStillOpen, but we now explicitly commit the TX on all tests, which is what they should be doing before. "closeTX" only does something if we exit the test before the commit, which means t.Fatalf was already called (or we had an early return, for example).